tests/visco_plastic_adiabatic_heating_density.prm

```diff
@@ -0,0 +1,113 @@
+# due to the increase in adiabatic temperature with depth.
+
+# Global parameters
+set Dimension                              = 2
+set Start time                             = 0
+set End time                               = 0
+set Use years in output instead of seconds = true
+set Nonlinear solver scheme                = single Advection, iterated Stokes
+set Max nonlinear iterations               = 1
+set Surface pressure                       = 0
+set Adiabatic surface temperature          = 1573
+
+# In combination with simplified adiabatic and shear heating,
+# the parameters give the Extended Boussinesq approximation.
+subsection Formulation
+  set Formulation          = custom
+  set Mass conservation    = incompressible
+  set Temperature equation = reference density profile
+end
+
+subsection Heating model
+  set List of model names = shear heating, adiabatic heating
+  subsection Adiabatic heating
+    set Use simplified adiabatic heating = true
+  end
+end
+
+# Model geometry (100x100 km, 100 km spacing)
+subsection Geometry model
+  set Model name = box
+  subsection Box
+    set X extent      = 100e3
+    set Y extent      = 100e3
+  end
+end
+
+# No mesh refinement
+subsection Mesh refinement
+  set Initial adaptive refinement        = 0
+  set Initial global refinement          = 0
+  set Time steps between mesh refinement = 0
+end
+
+# All boundaries have a fixed temperature
+subsection Boundary temperature model
+  set Fixed temperature boundary indicators  = bottom, top, left, right
+  set Model name = initial temperature
+end
+
+# Note that the initial temperature is the same as the adiabatic surface temperature
+subsection Initial temperature model
+  set Model name = function
+  subsection Function
+    set Function expression = 1573
+  end
+end
+
+subsection Boundary velocity model
+  set Tangential velocity boundary indicators = bottom, top, left, right
+end
+
+# Material model
+subsection Material model
+  set Model name = visco plastic
+  subsection Visco Plastic
+
+    # Since adiabatic heating is used, the reference temperature
+    # specified below will be ignored and the density formulation
+    # will use the adiabatic temperature.
+    set Reference temperature = 273
+    set Thermal expansivities = 1e-5
+    set Densities             = 3300
+
+    # Constant viscosity
+    set Viscous flow law                         = diffusion
+    set Grain size exponents for diffusion creep = 0.
+    set Prefactors for diffusion creep           = 5.e-22
+    set Activation energies for diffusion creep  = 0.
+    set Activation volumes for diffusion creep   = 0.
+  end
+end
+
+# Gravity model
+subsection Gravity model
+  set Model name = vertical
+  subsection Vertical
+    set Magnitude = 10.0
+  end
+end
+
+# Post processing
+subsection Postprocess
+  set List of postprocessors = mass flux statistics, material statistics, temperature statistics, visualization
```

Comment:
Doesn't really matter, but the mass flux statistics won't tell us much as the boundaries are all free slip. Maybe remove?